pkgs/build-support/rust/import-cargo-lock.nix

```diff
@@ -89,18 +93,23 @@ let
     '';
     fetchurl {
       name = "crate-${pkg.name}-${pkg.version}.tar.gz";
-      url = "https://crates.io/api/v1/crates/${pkg.name}/${pkg.version}/download";
+      url = "${downloadUrl}/v1/crates/${pkg.name}/${pkg.version}/download";
       sha256 = checksum;
     };
 
+  registries = {
+    "https://github.com/rust-lang/crates.io-index" = "https://crates.io/api";
```

Comment:
I thought about it again, maybe we should use `https://crates.io/api/v1/crates` instead, as documented [here](https://doc.rust-lang.org/cargo/reference/registry-index.html#index-configuration). What do you think?

Reply:
Probably more consistent indeed, I'll send a follow-up

Reply:
#222238